log/zap/tracer_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zap
+
+import (
+	"github.com/golang/mock/gomock"
```

Comment:
I strongly prefer not to introduce mock dependencies into this library. The goal of the logging wrapper is to log, the tests should be checking for log content, rather than mock expectations. Zap provides very good testing facilities, including checking for structured JSON.

Reply:
The tests in this file already verify the log message and in cases where appropriate whether the logger contains the specific log field. e.g., `context`. They don't look at the contents of the the key because that logic is tested in `field_test.go`. 

I agree that the purpose of the logging wrapper is to log - but it is crucial that it forwards calls and responses to and from the delegate. I think it is important to test this functionality. 

I don't see a way of using purely the logs in the loggingWrapper to verify that the delegate methods have been called with the correct arguments. What are you proposing?

Reply:
ok